core/dbt/task/retry.py

```diff
@@ -57,59 +62,61 @@
 
 class RetryTask(ConfiguredTask):
     def __init__(self, args, config, manifest) -> None:
-        super().__init__(args, config, manifest)
-
-        state_path = self.args.state or self.config.target_path
-
-        if self.args.warn_error:
-            RETRYABLE_STATUSES.add(NodeStatus.Warn)
-
-        self.previous_state = PreviousState(
-            state_path=Path(state_path),
-            target_path=Path(self.config.target_path),
-            project_root=Path(self.config.project_root),
+        # load previous run results
+        state_path = args.state or config.target_path
+        self.previous_results = load_result_state(
+            Path(config.project_root) / Path(state_path) / "run_results.json"
         )
-
-        if not self.previous_state.results:
+        if not self.previous_results:
             raise DbtRuntimeError(
                 f"Could not find previous run in '{state_path}' target directory"
             )
-
-        self.previous_args = self.previous_state.results.args
+        self.previous_args = self.previous_results.args
         self.previous_command_name = self.previous_args.get("which")
-        self.task_class = TASK_DICT.get(self.previous_command_name)  # type: ignore
-
-    def run(self):
-        unique_ids = set(
-            [
-                result.unique_id
-                for result in self.previous_state.results.results
-                if result.status in RETRYABLE_STATUSES
-            ]
-        )
 
-        cli_command = CMD_DICT.get(self.previous_command_name)
+        # Reslove flags and config
+        if args.warn_error:
+            RETRYABLE_STATUSES.add(NodeStatus.Warn)
 
+        cli_command = CMD_DICT.get(self.previous_command_name)  # type: ignore
         # Remove these args when their default values are present, otherwise they'll raise an exception
         args_to_remove = {
             "show": lambda x: True,
             "resource_types": lambda x: x == [],
             "warn_error_options": lambda x: x == {"exclude": [], "include": []},
         }
-
         for k, v in args_to_remove.items():
             if k in self.previous_args and v(self.previous_args[k]):
                 del self.previous_args[k]
-
         previous_args = {
-            k: v for k, v in self.previous_args.items() if k not in OVERRIDE_PARENT_FLAGS
+            k: v for k, v in self.previous_args.items() if k not in IGNORE_PARENT_FLAGS
+        }
+        click_context = get_current_context()
+        current_args = {
+            k: v
+            for k, v in args.__dict__.items()
+            if k in IGNORE_PARENT_FLAGS
+            or click_context.get_parameter_source(k) == ParameterSource.COMMANDLINE
```

Comment:
This is a key change, any parameter from the command line would overwrite the previous set parameter.
If it is not from command line and not in the `IGNORE_PARENT_FLAGS`, we will reuse things from previous invocation

Reply:
@ChenyuLInx Doesn't this mean that any parameter supplied on the command line would be used? That seems different than what we discussed with Grace, which was that only a limited set of parameters could be overriden when using retry.

Reply:
Thanks for calling it out. Updated.

Reply:
The code now looks better to me, but I think it also means we need to add some more parameters to the ALLOW_OVERRIDE_ARGS list. In particular the --target and --profile flags are overridden in production usage of dbt, as you can see using [this Datadog query](https://dbtlabsmt.datadoghq.com/logs?query=%22dbt%20client%20called%20with%22%20retry%20&cols=host%2Cservice&index=%2A&messageDisplay=inline&refresh_mode=sliding&stream_sort=desc&viz=stream&from_ts=1704403160572&to_ts=1704404060572&live=true).

Reply:
A [different Datadog query](https://dbtlabsmt.datadoghq.com/logs?query=%22cli%20with%20the%20following%20command%22%20retry%20&cols=host%2Cservice&event=AgAAAYzWaiMFv2_IbAAAAAAAAAAYAAAAAEFZeldhaWo5QUFCVVpWR3pLS2hZRWdBTwAAACQAAAAAMDE4Y2Q2NmItN2UzZi00NGJhLTgzNWUtYTk5OWMwMmNkZTI1&index=%2A&messageDisplay=inline&refresh_mode=sliding&stream_sort=desc&viz=stream&from_ts=1704403160572&to_ts=1704404060572&live=true) shows that --log-format, --debug, and --profiles-dir are also specified sometimes in production.

Reply:
So we actually have two list now, and they behave slightly different.
There's a `IGNORE_PARENT_FLAGS` whose behavior is: no matter what the last command set, we actually don't try to reuse them, but just use the new ones,
then there is the `ALLOW_OVERRIDE_ARGS`, which is : use whatever the command set, unless user specify something new in the retry command.

Is the name not obvious enough? should I change to different name?

Reply:
I think it would be slightly clearer if we called it ALLOW_CLI_OVERRIDE_FLAGS, but upon further review this looks okay.

Reply:
Updated!